airflow/models.py

```diff
@@ -335,10 +339,17 @@ def process_file(self, filepath, only_if_updated=True, safe_mode=True):
                         dag.full_filepath = filepath
                         if dag.fileloc != filepath:
                             dag.fileloc = filepath
-                    dag.is_subdag = False
-                    self.bag_dag(dag, parent_dag=dag, root_dag=dag)
-                    found_dags.append(dag)
-                    found_dags += dag.subdags
+                    try:
+                        dag.is_subdag = False
+                        self.bag_dag(dag, parent_dag=dag, root_dag=dag)
+                        found_dags.append(dag)
+                        found_dags += dag.subdags
+                    except Exception as e:
```

Comment:
Doesn't this happen somewhere else before?

Reply:
If you're referring to the exception handling, yes this is the same exception handling as a few lines above:
https://github.com/apache/incubator-airflow/pull/3116/files#diff-a32a363fa616685db3bfefba947535b2R300

Prior to this PR, the cycle detection exception was raised immediately when a task was added to a DAG and an error would be thrown directly when the dag was imported: https://github.com/apache/incubator-airflow/pull/3116/files#diff-a32a363fa616685db3bfefba947535b2R300 

Now with this PR, the cycle detection exception is raised when bag_dag is called (i.e. before a dag can be added to the dagbag).

Is that what you're referring to?

Reply:
The exception is really broad, can we make this more specific?

Reply:
Yes, added new `AirflowDagCycleException`